libs/vm-modules/src/core/structured_data.cpp

```diff
@@ -39,8 +39,173 @@ namespace vm_modules {
 namespace {
 
 using fetch::byte_array::ConstByteArray;
-using fetch::byte_array::ToBase64;
-using fetch::byte_array::FromBase64;
+using fetch::byte_array::ByteArray;
+using fetch::vm_modules::ByteArrayWrapper;
+using fetch::vm_modules::math::UInt256Wrapper;
+
+template <typename T>
+constexpr uint8_t getTypeId() noexcept;
+
+template <>
+constexpr uint8_t getTypeId<vm::String>() noexcept
+{
+  return 's';
```

Comment:
I am not really sure that I like the idea of having yet another internal serialisations format. I would have preferred that we keep structured data as a loose wrapper around our MsgPack format and that we come up with a concrete ABI design that can be stared for actions, queries state etc.

Reply:
Yes I fully agree. I knew from the very beginning that this is idiom going to be contentious.